lib/cylc/task_pool.py

```diff
@@ -1263,16 +1263,13 @@ def get_task_requisites(self, items, list_prereqs=False):
                 else:
                     extras['External trigger "%s"' % trig] = 'NOT satisfied'
             for label, satisfied in itask.state.xtriggers.items():
+                extra = 'xtrigger "%s = %s"' % (
+                    label, self.xtrigger_mgr.get_xtrig_ctx(
+                        itask, label).get_signature())
                 if satisfied:
-                    extras['xtrigger "%s"' % label] = 'satisfied'
+                    extras[extra] = 'satisfied'
                 else:
-                    extras['xtrigger "%s"' % label] = 'NOT satisfied'
-            if itask.state.xclock is not None:
-                label, satisfied = itask.state.xclock
-                if satisfied:
-                    extras['xclock "%s"' % label] = 'satisfied'
-                else:
-                    extras['xclock "%s"' % label] = 'NOT satisfied'
+                    extras[extra] = 'NOT satisfied'
```

Comment:
Not the concern of this change, so maybe a follow-on refactor.

I guess we are leaking presentation layer logic into the server. I would be much happier if we have `extras[extra] = satisfied` here (the value is presumably a boolean or None), and let the presentation layer (e.g. `cylc show`) to decide the message string (or icon or whatever).

Reply:
I agree, and this particular bit doesn't work on master, which (nicely) has decouple xtrigger_mgr from this module.  I'll think about it.

Reply:
I've done a small refactor on master to move this whole method out of the task_pool module (will push it to #3287 shortly).  No need to do it on 7.8.x though.